File: evidence/generated-systems/angular/fintech-on-angular/src/app/models/FinancialInstitution.js

// Define collection and schema for FinancialInstitution
export interface FinancialInstitution {
    name:
	type : string
    legalName:
	type : string
    countryOfIncorporation:
	type : string
    bic:
	type : BIC
    website:
	type : string
    Branches:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Branch' }]
    Customers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Customer' }]
    ProductOfferings:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ProductOffering' }]
    PaymentProcessors:
 	type : [{ type: Schema.Types.ObjectId, ref: 'PaymentProcessor' }]
    CompliancePolicies:
 	type : [{ type: Schema.Types.ObjectId, ref: 'CompliancePolicy' }]
#
    collection: 'financialInstitutions'
}

File: evidence/generated-systems/django/fintech-on-django/fintechOnDjango/models/FinancialInstitution.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model FinancialInstitution
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class FinancialInstitution Declaration
#======================================================================
class FinancialInstitution (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	legalName = models.CharField(max_length=200, null=True)
	countryOfIncorporation = models.CharField(max_length=200, null=True)
	bic = BIC
	website = models.CharField(max_length=200, null=True)
	branches = models.ManyToManyField('Branch',  blank=True, related_name='+')
	customers = models.ManyToManyField('Customer',  blank=True, related_name='+')
	productOfferings = models.ManyToManyField('ProductOffering',  blank=True, related_name='+')
	paymentProcessors = models.ManyToManyField('PaymentProcessor',  blank=True, related_name='+')
	compliancePolicies = models.ManyToManyField('CompliancePolicy',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.legalName
		str = str + self.countryOfIncorporation
		str = str + self.bic
		str = str + self.website
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "FinancialInstitution";
    
	def objectType(self):
		return "FinancialInstitution";
